modules/xfem/src/userobjects/ComboCutUserObject.C

```diff
@@ -0,0 +1,119 @@
+  : GeometricCutUserObject(parameters),
+    _cut_names(getParam<std::vector<UserObjectName>>("cuts")),
+    _num_cuts(_cut_names.size()),
+    _cuts(_num_cuts),
+    _keys(getParam<std::vector<GeometricCutSubdomainID>>("keys")),
+    _vals(getParam<std::vector<GeometricCutSubdomainID>>("vals"))
+{
+  for (unsigned int i = 0; i < _num_cuts; i++)
+    _cuts[i] = &getUserObjectByName<GeometricCutUserObject>(_cut_names[i]);
+
+  buildMap();
+}
+
+void
+ComboCutUserObject::buildMap()
+{
+  if (_keys.size() % _num_cuts != 0)
+    mooseError("Expect multiples of ", _num_cuts, " keys, but got ", _keys.size(), " keys.");
+
+  unsigned int num_combos = _keys.size() / _num_cuts;
+
+  if (_vals.size() != num_combos)
+    mooseError("Expect one value for each of the combination, got ",
+               num_combos,
+               " combinations, but got ",
+               _vals.size(),
+               " values.");
+
+  for (unsigned int i = 0; i < num_combos; i++)
+  {
+    std::vector<GeometricCutSubdomainID> combo_key;
+    for (unsigned int j = 0; j < _num_cuts; j++)
+      combo_key.push_back(_keys[i * _num_cuts + j]);
+    _combo_ids.emplace(combo_key, _vals[i]);
+  }
+}
+
+bool
+ComboCutUserObject::cutElementByGeometry(const Elem * elem,
+                                         std::vector<Xfem::CutEdge> & cut_edges,
+                                         std::vector<Xfem::CutNode> & cut_nodes,
+                                         Real time) const
+{
+  for (auto cut : _cuts)
+    if (cut->cutElementByGeometry(elem, cut_edges, cut_nodes, time))
+      return true;
+  return false;
+}
+
+bool
+ComboCutUserObject::cutElementByGeometry(const Elem * elem,
+                                         std::vector<Xfem::CutFace> & cut_faces,
+                                         Real time) const
+{
+  for (auto cut : _cuts)
+    if (cut->cutElementByGeometry(elem, cut_faces, time))
+      return true;
+  return false;
+}
+
+bool
+ComboCutUserObject::cutFragmentByGeometry(std::vector<std::vector<Point>> & frag_edges,
+                                          std::vector<Xfem::CutEdge> & cut_edges,
+                                          Real time) const
+{
+  for (auto cut : _cuts)
+    if (cut->cutFragmentByGeometry(frag_edges, cut_edges, time))
+      return true;
+  return false;
+}
+
+bool
+ComboCutUserObject::cutFragmentByGeometry(std::vector<std::vector<Point>> & frag_faces,
+                                          std::vector<Xfem::CutFace> & cut_faces,
+                                          Real time) const
+{
+  for (auto cut : _cuts)
+    if (cut->cutFragmentByGeometry(frag_faces, cut_faces, time))
+      return true;
+  return false;
+}
+
+GeometricCutSubdomainID
+ComboCutUserObject::getGeometricCutSubdomainID(const Node * node) const
+{
+  std::vector<GeometricCutSubdomainID> combo_key;
+  for (auto cut : _cuts)
+    combo_key.push_back(cut->getGeometricCutSubdomainID(node));
+  return _combo_ids.at(combo_key);
```

Comment:
It seems like there's a chance that a particular combination might not exist, so we might need to handle that error. I'm saying that because I don't think the check on line 46 of this class exhaustively checks that you have every combination covered.

Reply:
Yeah, that check in `GeometricCutSwitchingMaterial.C` should be moved here.